File: _API_TESTING_GHERKIN_/DEMOAPP003_TYPESCRIPT_PLAYWRIGHT/screenplay/questions/ResponseStatus.ts
import type { Question } from "../support/types";
import type { Actor } from "../actors/Actor";
import type { APIResponse } from "@playwright/test";

export class ResponseStatus implements Question<number> {
  static code(): ResponseStatus {
    return new ResponseStatus();
  }

  answeredBy(actor: Actor): number {
    const response = actor.recall<APIResponse>("last-response");
    if (!response) {
      throw new Error("No response available for status assertion");
    }
    return response.status();
  }
}

File: _API_TESTING_GHERKIN_/DEMOAPP004_PYTHON_PLAYWRIGHT/screenplay/questions/response_status.py
"""Question that returns the last response status code."""

from playwright.sync_api import APIResponse

from screenplay.support.memory_keys import MemoryKeys


class ResponseStatus:
    @staticmethod
    def answered_by(actor) -> int:
        response: APIResponse = actor.memory.recall(MemoryKeys.LAST_RESPONSE)
        if response is None:
            raise AssertionError("No response stored in memory")
        return response.status
